src/redux/requests.ts

```diff
@@ -0,0 +1,282 @@
+import { filter, get, omit, values } from 'lodash';
+import { Dispatch } from 'redux';
+import { AppGetState } from './store';
+import { maybeSignUri } from '@rainbow-me/handlers/imgix';
+import {
+  getLocalRequests,
+  removeLocalRequest,
+  saveLocalRequests,
+} from '@rainbow-me/handlers/localstorage/walletconnectRequests';
+import {
+  dappLogoOverride,
+  dappNameOverride,
+} from '@rainbow-me/helpers/dappNameHandler';
+import { getRequestDisplayDetails } from '@rainbow-me/parsers';
+import { ethereumUtils } from '@rainbow-me/utils';
+import logger from 'logger';
+
+// -- Constants --------------------------------------- //
+
+const REQUESTS_UPDATE_REQUESTS_TO_APPROVE =
+  'requests/REQUESTS_UPDATE_REQUESTS_TO_APPROVE';
+const REQUESTS_CLEAR_STATE = 'requests/REQUESTS_CLEAR_STATE';
+
+// Requests expire automatically after 1 hour
+const EXPIRATION_THRESHOLD_IN_MS = 1000 * 60 * 60;
+
+// -- Actions ----------------------------------------- //
+
+/**
+ * A request stored in state.
+ */
+interface RequestData {
+  /**
+   * The WalletConnect client ID for the request.
+   */
+  clientId: string;
+
+  /**
+   * The WalletConnect peer ID for the request.
+   */
+  peerId: string;
+
+  /**
+   * The request ID.
+   */
+  requestId: number;
```

Comment:
is `requestId` a number and not a string? the `RequestsState` (line 103) has a mapping with `requestId` as string

Reply:
@jinchung As I changed the mapping to use a number instead. As far as I can tell, this `requestId` corresponds to an `id` from WalletConnect based on https://github.com/rainbow-me/rainbow/blob/f159380de466658ef22ad475611b58d9683ab505/src/redux/walletconnect.js#L320. The WalletConnect documentation implies that that's a number, for instance [here](https://github.com/WalletConnect/walletconnect-docs/blob/e992a83e2bb02065f7a7235b019917ed0e9f8bbe/versioned_docs/version-1.0/quick-start/wallets/react-native.md).